src/Transport/Connectivity/MessageReceiverLifeCycleManager.cs

```diff
@@ -20,13 +20,10 @@ public IMessageReceiver Get(string entityPath, string namespaceAlias)
             var buffer = MessageReceivers.GetOrAdd(entityPath + namespaceAlias, s =>
             {
                 var b = new CircularBuffer<EntityClientEntry>(numberOfReceiversPerEntity);
-                for (var i = 0; i < numberOfReceiversPerEntity; i++)
+                for(var i = 0; i < numberOfReceiversPerEntity; i++)
                 {
                     var e = new EntityClientEntry();
-                    lock (e.Mutex)
-                    {
-                        e.ClientEntity = receiveFactory.Create(entityPath, namespaceAlias).GetAwaiter().GetResult();
-                    }
+                    e.ClientEntity = receiveFactory.Create(entityPath, namespaceAlias).GetAwaiter().GetResult();
```

Comment:
Since `MessageReceivers` is a concurrent dictionary, lock is not needed I presume @yvesgoeleven